Fixes #3906: Incorrect output for rich text in hints and solution

vrajdesai78 opened the pull request:
## Explanation
Fixes #3906: Fixed the overall accessibility of Hints and Solution as follows:
- Parse HTML to normal string for content description
- Improved flow of screen reader by removing repetitive content description.
- Changed content description of expand_list_icon and also improve the flow by making expand_list_icon clickable and not clickable based on talkback reader. 

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### Video for Hints


https://user-images.githubusercontent.com/43074241/198126083-88b1ea29-3726-43e5-895e-de64166a6515.mp4


### Video for Solution


https://user-images.githubusercontent.com/43074241/198126121-59b56e5c-4003-4047-a6c9-4595430f3a8b.mp4

vrajdesai78 committed:
Fixed htmlcontent in contentDescription

vrajdesai78 committed:
Optimized Unused Imports

vrajdesai78 committed:
Merge branch 'develop' of https://github.com/vrajdesai78/oppia-android into Incorrect-output-for-rich-text-in-Hints-and-Solution

vrajdesai78 committed:
Rearranged code

vrajdesai78 commented:
@BenHenning @rt4914 CI checks are failing due to some flaky tests. Can you re-run it.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt:
```diff
@@ -1617,6 +1617,48 @@ class ExplorationActivityTest {
 
+  @Test
+  @RunOn(TestPlatform.ROBOLECTRIC) // TODO(#3858): Enable for Espresso.
+  fun testExpActivity_showHintAndSolution_hasCorrectContentDescription() {
```
Just to confirm, does this fail without the fix?

vrajdesai78 replied:
Yes @BenHenning, test case is failing without parsing HTML. 
![image](https://user-images.githubusercontent.com/43074241/196857392-84f7d1a9-37ef-41ac-a978-5d9fee1fa1fe.png)

BenHenning replied:
Thanks for checking!

BenHenning reviewed (changes requested):
Thanks @vrajdesai78. Just had one follow-up comment--PTAL.

vrajdesai78 commented:
> Thanks @vrajdesai78. Just had one follow-up comment--PTAL.

@BenHenning replied to your comment, also can you please re-run CI checks. Thanks

BenHenning reviewed (approved):
Thanks @vrajdesai78! This LGTM. Just waiting on CI.

BenHenning commented:
CI is looking good, so I have no other concerns here. I think you just need @rt4914's approval at this point.

rt4914 reviewed (changes requested):
Currently the hint-content and "Hint 1", `show/hide` all these are getting read-at-once.

Instead the hint-content, ie., `Remember ....` should be entirely separate from other things like `Hint 1` and `show/hide`

vrajdesai78 committed:
seperated content description

vrajdesai78 committed:
Optimized code

vrajdesai78 commented:
> Currently the hint-content and "Hint 1", `show/hide` all these are getting read-at-once.
> 
> Instead the hint-content, ie., `Remember ....` should be entirely separate from other things like `Hint 1` and `show/hide`

Thanks @rt4914, I have updated code according to your suggestion. Also, updated video in the description.

vrajdesai78 committed:
updated strings.xml

rt4914 commented on app/src/main/res/layout/hints_summary.xml:
```diff
@@ -94,7 +95,7 @@
             android:layout_marginTop="8dp"
             android:layout_marginEnd="8dp"
             android:layout_marginBottom="4dp"
-            android:contentDescription="@{viewModel.computeHintListDropDownIconContentDescription()}"
```
One more suggestion, 

use `ImageButton` instead of `ImageView`, so that the talkback will correctly read the output for button instead of `Image`

Also, why was the content description should be optimised, i.e.,
- if list is expanded, 'Hide hint'
- if list is not-expanded, 'Show hint', 

**Also, note, I think similar changes needs to be done for `Solution-summary` too, please check that too..**

vrajdesai78 replied:
I have made required changes for Solution-summary. Updated changes and video in PR's description. Thanks

rt4914 reviewed (changes requested):
@vrajdesai78 Suggested some more changes based on the current output.

vrajdesai78 committed:
Accessibility flow fixed for solution also

vrajdesai78 committed:
Optimized code

rt4914 reviewed (approved):
LGTM, Thanks.
